base/uk.ac.stfc.isis.ibex.ui.configserver.tests/src/uk/ac/stfc/isis/ibex/ui/configserver/tests/CheckboxLabelProviderTest.java

```diff
@@ -0,0 +1,223 @@
+ /*
+ * This file is part of the ISIS IBEX application.
+ * Copyright (C) 2012-2016 Science & Technology Facilities Council.
+ * All rights reserved.
+ *
+ * This program is distributed in the hope that it will be useful.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v1.0 which accompanies this distribution.
+ * EXCEPT AS EXPRESSLY SET FORTH IN THE ECLIPSE PUBLIC LICENSE V1.0, THE PROGRAM 
+ * AND ACCOMPANYING MATERIALS ARE PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES 
+ * OR CONDITIONS OF ANY KIND.  See the Eclipse Public License v1.0 for more details.
+ *
+ * You should have received a copy of the Eclipse Public License v1.0
+ * along with this program; if not, you can obtain a copy from
+ * https://www.eclipse.org/org/documents/epl-v10.php or 
+ * http://opensource.org/licenses/eclipse-1.0.php
+ */
+
+/**
+ * 
+ */
+package uk.ac.stfc.isis.ibex.ui.configserver.tests;
+
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
+
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.eclipse.core.databinding.observable.map.IObservableMap;
+import org.eclipse.core.databinding.observable.map.ObservableMap;
+import org.eclipse.jface.viewers.TableViewer;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionListener;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Listener;
+import org.eclipse.swt.widgets.TypedListener;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import uk.ac.stfc.isis.ibex.ui.configserver.CheckboxLabelProvider;
+import uk.ac.stfc.isis.ibex.ui.configserver.CheckboxLabelProvider.CheckboxSelectionAdapter;
+import uk.ac.stfc.isis.ibex.ui.tables.DataboundTable;
+
+/**
+ *
+ */
+public class CheckboxLabelProviderTest {
```

Comment:
First of all, updateCheckboxListenerUpdateFlags has nothing to do with the listeners.

Secondly, I had to make some methods public because I could not use update() directly. This is because I would need to do a lot of mocking of UI elements in order to mock getControl(), and I was told I should not mock a method of the class I am testing, since then I am testing a mock, and not the real thing.

Thidly, I do not see how my tests are too specific to the implementation. The 4 tests for the flag map are for general situations, like adding new elements, removing old ones, making sure the map is unchanged if you call update again, and making sure you can reset all of the flags. Do you mean that my test names are a bit too specific?

Reply:
On further discussion , I see what Dom meant. Changed tests to make more clean and simple.